openpgp/read.go

```diff
@@ -270,13 +272,17 @@ FindLiteralData:
 				prevLast = true
 			}
 
-			h, wrappedHash, err = hashForSignature(p.Hash, p.SigType)
+			h, wrappedHash, err = hashForSignature(p.Hash, p.SigType, p.Salt)
 			if err != nil {
 				md.SignatureError = err
 			}
 
 			md.IsSigned = true
+			if p.Version == 6 {
+				md.SignedByFingerprint = p.KeyFingerprint
+			} 
```

Comment:
I don't think this property is named accurately; this is the fingerprint of the key the message was encrypted to, not the key it was signed by.

Reply:
I think it should be correct. It is just the fingerprint of the OnePassSignature packet which contains the fingerprint of the signing key.

Reply:
Ah sorry, I misread and thought `p` was referring to the PKESK at this point